source.go

```diff
@@ -15,81 +38,209 @@ type Source struct {
-	// SourceConfigParam is named foo and must be provided by the user.
-	SourceConfigParam string `json:"foo" validate:"required"`
+	// Table is the DynamoDB table name to pull data from.
+	Table string `json:"table" validate:"required"`
+	// AWS region.
+	AWSRegion string `json:"aws.region" validate:"required"`
+	// AWS access key id.
+	AWSAccessKeyID string `json:"aws.accessKeyId" validate:"required"`
+	// AWS secret access key.
+	AWSSecretAccessKey string `json:"aws.secretAccessKey" validate:"required"`
+	// polling period for the CDC mode, formatted as a time.Duration string.
+	PollingPeriod time.Duration `json:"pollingPeriod" default:"1s"`
+
+	partitionKey string
+	sortKey      string
+}
+
+type Iterator interface {
+	HasNext(ctx context.Context) bool
+	Next(ctx context.Context) (opencdc.Record, error)
+	Stop()
 }
 
 func NewSource() sdk.Source {
-	// Create Source and wrap it in the default middleware.
 	return sdk.SourceWithMiddleware(&Source{}, sdk.DefaultSourceMiddleware()...)
 }
 
-func (s *Source) Parameters() config.Parameters {
-	// Parameters is a map of named Parameters that describe how to configure
-	// the Source. Parameters can be generated from SourceConfig with paramgen.
+func (s *Source) Parameters() cconfig.Parameters {
 	return s.config.Parameters()
 }
 
-func (s *Source) Configure(ctx context.Context, cfg config.Config) error {
-	// Configure is the first function to be called in a connector. It provides
-	// the connector with the configuration that can be validated and stored.
-	// In case the configuration is not valid it should return an error.
-	// Testing if your connector can reach the configured data source should be
-	// done in Open, not in Configure.
-	// The SDK will validate the configuration and populate default values
-	// before calling Configure. If you need to do more complex validations you
-	// can do them manually here.
-
-	sdk.Logger(ctx).Info().Msg("Configuring Source...")
-	err := sdk.Util.ParseConfig(ctx, cfg, &s.config, NewSource().Parameters())
+func (s *Source) Configure(ctx context.Context, cfg cconfig.Config) error {
+	sdk.Logger(ctx).Info().Msg("Configuring DynamoDB Source...")
+	var sourceConfig SourceConfig
+	err := sdk.Util.ParseConfig(ctx, cfg, &sourceConfig, NewSource().Parameters())
 	if err != nil {
 		return fmt.Errorf("invalid config: %w", err)
 	}
+	s.config = sourceConfig
+
 	return nil
 }
 
-func (s *Source) Open(_ context.Context, _ opencdc.Position) error {
-	// Open is called after Configure to signal the plugin it can prepare to
-	// start producing records. If needed, the plugin should open connections in
-	// this function. The position parameter will contain the position of the
-	// last record that was successfully processed, Source should therefore
-	// start producing records after this position. The context passed to Open
-	// will be cancelled once the plugin receives a stop signal from Conduit.
+func (s *Source) Open(ctx context.Context, pos opencdc.Position) error {
```

Comment:
yeah this is one of my todos, CDC needs some improvements like detecting how much time since the last run, start capturing events while snapshot is running, and continue from a certain position. Some of these need the timestamp in the position so it will take some time but has to be done. (this version is a quick one for the demo)

Reply:
a simpler approach here could be to always warn the user when the pipeline is restarted! let them know that if more than 24h has passed then any events that happened before 24h will be lost. wdyt? I'm still trying to decide weather I'm gonna need the timestamp in the position or not, will have more details once I have the CDC position handling figured

Reply:
IMHO, it would be ideal if we could detect the time when the connector stopped. Again, ideally, we would get that through Conduit, because saving the timestamp in a position isn't quite correct (for example, some time could have passed between the last record and the connector being stopped). So yeah, looks like logging is the best thing we can do for now. Once we get connector storage we can try finding a better solution.

Reply:
I added this in the CDC iterator, if the sequence number is not found that means it was expired, so I log a warning